_aicspylibczi/Reader.cpp

```diff
@@ -285,6 +285,22 @@ namespace pylibczi {
       return metaSubblocks;
   }
 
+  libCZI::IntRect
+  Reader::readSubblockRect(libCZI::CDimCoordinate& plane_coord_, int index_m_)
+  {
+      SubblockSortable subBlockToFind(&plane_coord_, index_m_, isMosaic());
+      SubblockIndexVec matches = getMatches(subBlockToFind);
```

Comment:
You're retrieving a vector of matches. Why loop through the list and return the last one?
If the length of the list != 1 then the plane_coord_ was underspecified? Maybe throw an exception in that case or?

Reply:
You are right that the loop is unnecessary, as all matches should produce the same result if not misspecified. And you right that proper specification should be checked (as you wrote in your other comment).